tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -0,0 +1,435 @@
+  val DEFAULT_MEMORY_OVERHEAD_FACTOR: Double = 0.1
+  val MIN_MEMORY_OVERHEAD = "2g"
+  val MAX_EXTRA_SYSTEM_MEMORY = "2g"
+
+  val MAX_PER_EXECUTOR_CORE_COUNT: Int = 16
+  val MIN_PER_EXECUTOR_CORE_COUNT: Int = 4
+
+  val MAX_EXECUTOR_MEMORY: String = "64g"
+  val MIN_EXECUTOR_MEMORY: String = "8g"
+
+  var comments: Seq[String] = Seq()
+
+  private def recommendSparkProperties(bestConfig: Config, systemProps: SystemProps): Unit = {
+    if (systemProps == null) {
+      logWarning("System information is not available. Cannot recommend properties.")
+      comments :+= "'spark.executor.memory' should be set to at least 2GB/core."
+      comments :+= "'spark.executor.instances' should be set to 'num_gpus * num_workers'."
+    } else {
+      // Recommendation for 'spark.executor.instances' based on number of gpus and workers
+      systemProps.num_workers match {
+        case Some(numWorkers) =>
+          val numInstances = if (systemProps.gpu_props != null) {
+            numWorkers * systemProps.gpu_props.count
+          } else numWorkers
+          bestConfig.set_executor_instances(numInstances)
+        case None =>
+          val num_gpus_str = if (systemProps.gpu_props != null) {
+            systemProps.gpu_props.count.toString
+          } else "num_gpus"
+
+          comments :+= s"'spark.executor.instances' should be set to $num_gpus_str * num_workers."
+      }
+
+      // Recommendation for 'spark.executor.cores' based on number of cpu cores and gpus
+      val numCores: Int = if (systemProps.gpu_props != null) {
+        Math.min(systemProps.numCores * 1.0 / systemProps.gpu_props.count,
+          MAX_PER_EXECUTOR_CORE_COUNT).toInt
+      } else systemProps.numCores
+
+      if(numCores < MIN_PER_EXECUTOR_CORE_COUNT) {
+        comments :+= s"Number of cores per executor is very low. " +
+          s"It is recommended to have at least $MIN_PER_EXECUTOR_CORE_COUNT cores per executor."
+      }
+
+      if (systemProps.num_workers.nonEmpty) {
+        val numInstances = bestConfig.get_executor_instances
+        if (numCores * numInstances < systemProps.numCores) {
+          comments :+= "Not all cores in the machine are being used. " +
+            "It is recommended to use different machine."
+        }
+      }
+
+      bestConfig.set_executor_cores(numCores)
+
+      // Recommendation for 'spark.executor.memory' based on system memory, cluster scheduler
+      // and num of gpus
+      val sparkMaster = getSparkProperty(app, "spark.master")
+      val systemMemoryNum: Long = convertFromHumanReadableSize(systemProps.memory)
+      val extraSystemMemoryNum: Long = convertFromHumanReadableSize(MAX_EXTRA_SYSTEM_MEMORY)
+      val effectiveSystemMemoryNum: Long =
+        if (sparkMaster.contains("yarn") || sparkMaster.contains("k8s")) {
+          systemMemoryNum -  extraSystemMemoryNum -
+            convertFromHumanReadableSize(MIN_MEMORY_OVERHEAD)
+        } else systemMemoryNum - extraSystemMemoryNum
```

Comment:
actually thinking about this some more yarn and k8s do configured meory differently then standalone but we still want to take into account pinned memory in the overall system size so we should probably make the effectiveSystemMemoryNum the same

Reply:
Made effective system memory same irrespective of cluster scheduler.